### PR TITLE
Refactor logic related to building of policy string

### PR DESCRIPTION
**Pull request by jelhan**
- Policy object values should not be converted to strings.
- Policy string should be calculated once and reused.
- Fixes a bug that CSP is delivered through header even if `delivery` does not include `"header"`.
- Improves readability of tests.

### CODE REVIEW

**Comment by jelhan** [comment]
Tests are failing due to https://github.com/tomdale/ember-cli-addon-tests/issues/212.

**Comment by jelhan** [comment]
Restarted the tests after `portfinder` has been fixed. Ready for review + merge from my side.

File: index.js [hunk 1]
```diff
@@ -50,66 +50,61 @@ module.exports = {
...
+
+    // Need to recalculate the policy if local development server is used to
+    // support live reload, executing tests in development enviroment via
+    // `http://localhost:4200/tests` and reporting CSP violations on CLI.
+    let policyObject = this._config.policy;
+
+    // the local server will never run for production builds, so no danger in
+    // adding the nonce all the time even so it's only needed if tests are
+    // executed by opening `http://localhost:4200/tests`
+    appendSourceList(policyObject, 'script-src', "'nonce-" + STATIC_TEST_NONCE + "'");
```
Comment (sandstrom) [inline index.js]: As an extra precaution, perhaps we should check `environment === 'test' || environment === 'development'`?

Comment (jelhan) [reply index.js]: I think the most precise test would be if `tests` is enabled. This seems to be controlled by `tests` property of `EmberApp`, which should be available as `app.tests` in `included` hook and maybe also as `this.app.tests` directly here. Let me double check.

Comment (jelhan) [reply index.js]: This was an awesome hint! :hugs: `app.tests` allows us to always add the nonce required for test support if and only if the build includes tests. This also supports edge cases like `ember test --environment production`.

This relies on the assumption that hopefully nobody deploys a build including tests. But if we don't do that assumption we would break test support for some environments. And this would also not give us that much coincidence cause someone might deploy a development build?!?

File: index.js [hunk 2]
```diff
@@ -244,19 +178,31 @@ module.exports = {
   // that one is executed after `serverMiddleware` and can't do it in `serverMiddleware`
   // hook cause that one is only executed on `ember serve` but not on `ember build` or
   // `ember test`. We can't do it in `init` hook cause app is not available by then.
+  //
+  // The same applies to policy string generation. It's also calculated in `included`
+  // hook and reused in both others. But this one might be overriden in `serverMiddleware`
+  // hook to support live reload. This is safe cause `serverMiddleware` hook is executed
+  // before `contentFor` hook.
   included: function(app) {
```
[the remark this answers is not in the document]
Comment (jelhan) [reply index.js]: We are not calling super from any hook. Is there anything special about `included` one?

[the remark this answers is not in the document]
Comment (rwjblue) [reply index.js]: Only some of the hooks have default implementations, those are the ones that you have to super from. `included` has a default implementation.

[the remark this answers is not in the document]
Comment (jelhan) [reply index.js]: :ok_hand: Added the super call in this PR.

**Comment by jelhan** [comment]
Addressed review feedback and rewritten the history for readability. Please feel free to merge as soon as you have a free minute.

**Comment by rwjblue** [comment]
Thanks @jelhan!
